### ISSUE TITLE
Add support for Huion Kamvas Pro 16 (2.5K)

### ISSUE THREAD

**Issue by DennisWG** [OpenTabletDriver/OpenTabletDriver#2390]
### Acknowledgements

- [X] I have searched the existing issues and this new issue is not a duplicate of any.
- [X] I have written a concise and meaningful title.

### Manufacturer

Huion

### Product Name

Kamvas Pro 16 (2.5K)

### Device Width

349.6

### Device Height

196.7

### Pressure Levels

8192

### Additional Information

The Huion Kamvas Pro 16 (2.5K) is not supported out-of-the-box. I got it working using the following custom configuration:

```json
{
  "Name": "Huion Kamvas Pro 16 (2.5K)",
  "Specifications": {
    "Digitizer": {
      "Width": 349.6,
      "Height": 196.7,
      "MaxX": 69926.0,
      "MaxY": 39333.0
    },
    "Pen": {
      "MaxPressure": 8191,
      "Buttons": {
        "ButtonCount": 2
      }
    },
    "AuxiliaryButtons": {
      "ButtonCount": 8
    },
    "MouseButtons": null,
    "Touch": null
  },
  "DigitizerIdentifiers": [
    {
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 12,
      "OutputReportLength": null,
      "ReportParser": "OpenTabletDriver.Configurations.Parsers.Huion.GianoReportParser",
      "FeatureInitReport": null,
      "OutputInitReport": null,
      "DeviceStrings": {
        "201": "HUION_M214_\\d{6}$"
      },
      "InitializationStrings": null
    }
  ],
  "AuxilaryDeviceIdentifiers": [],
  "Attributes": {}
}
```

I'm not too sure about `InitializationStrings`, `AuxilaryDeviceIdentifiers` and `Attributes` since I can't find any documentation regarding the configuration files. Also I've used the tablet's working-area for both `Width` and `Height`. This seems to be what the other configruations do aswell. Nevertheless this seems to be working just fine. 

I'm happy to provide any additional information as needed!

### Diagnostic Information

```json
{
  "App Version": "OpenTabletDriver v0.6.0.4",
  "Build Date": "5/4/2022",
  "Operating System": {
    "Platform": 2,
    "ServicePack": "",
    "Version": "10.0.19044.0",
    "VersionString": "Microsoft Windows NT 10.0.19044.0"
  },
  "HID Devices": [
    {
      "DevicePath": "\\\\?\\hid#vid_256c&pid_006d&mi_00&col01#7&310b9f2e&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}",
      "Manufacturer": "HUION",
      "ProductName": "Huion Tablet_GT1602",
      "SerialNumber": "",
      "FriendlyName": "Huion Tablet_GT1602",
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 12,
      "OutputReportLength": 0,
      "FeatureReportLength": 0,
      "CanOpen": true
    },
    {
      "DevicePath": "\\\\?\\hid#vid_256c&pid_006d&mi_00&col02#7&310b9f2e&0&0001#{4d1e55b2-f16f-11cf-88cb-001111000030}",
      "Manufacturer": "HUION",
      "ProductName": "Huion Tablet_GT1602",
      "SerialNumber": "",
      "FriendlyName": "Huion Tablet_GT1602",
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 0,
      "OutputReportLength": 0,
      "FeatureReportLength": 8,
      "CanOpen": true
    },
    {
      "DevicePath": "\\\\?\\hid#vid_256c&pid_006d&mi_01&col01#7&d34616c&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}",
      "Manufacturer": "HUION",
      "ProductName": "Huion Tablet_GT1602",
      "SerialNumber": "",
      "FriendlyName": "Huion Tablet_GT1602",
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 10,
      "OutputReportLength": 0,
      "FeatureReportLength": 0,
      "CanOpen": false
    },
    {
      "DevicePath": "\\\\?\\hid#vid_256c&pid_006d&mi_01&col02#7&d34616c&0&0001#{4d1e55b2-f16f-11cf-88cb-001111000030}",
      "Manufacturer": "HUION",
      "ProductName": "Huion Tablet_GT1602",
      "SerialNumber": "",
      "FriendlyName": "Huion Tablet_GT1602",
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 28,
      "OutputReportLength": 0,
      "FeatureReportLength": 257,
      "CanOpen": false
    },
    {
      "DevicePath": "\\\\?\\hid#vid_256c&pid_006d&mi_02&col01#7&24f7edb1&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}\\kbd",
      "Manufacturer": "HUION",
      "ProductName": "Huion Tablet_GT1602",
      "SerialNumber": "",
      "FriendlyName": "Huion Tablet_GT1602",
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 8,
      "OutputReportLength": 0,
      "FeatureReportLength": 0,
      "CanOpen": false
    },
    {
      "DevicePath": "\\\\?\\hid#vid_256c&pid_006d&mi_02&col03#7&24f7edb1&0&0002#{4d1e55b2-f16f-11cf-88cb-001111000030}",
      "Manufacturer": "HUION",
      "ProductName": "Huion Tablet_GT1602",
      "SerialNumber": "",
      "FriendlyName": "Huion Tablet_GT1602",
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 2,
      "OutputReportLength": 0,
      "FeatureReportLength": 0,
      "CanOpen": true
    },
    {
      "DevicePath": "\\\\?\\hid#vid_256c&pid_006d&mi_02&col02#7&24f7edb1&0&0001#{4d1e55b2-f16f-11cf-88cb-001111000030}",
      "Manufacturer": "HUION",
      "ProductName": "Huion Tablet_GT1602",
      "SerialNumber": "",
      "FriendlyName": "Huion Tablet_GT1602",
      "VendorID": 9580,
      "ProductID": 109,
      "InputReportLength": 3,
      "OutputReportLength": 0,
      "FeatureReportLength": 0,
      "CanOpen": true
    }
  ],
  "Console Log": [
    {
      "Time": "2022-08-21T02:21:41.7798065+02:00",
      "Group": "Detect",
      "Message": "Searching for tablets...",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7798123+02:00",
      "Group": "Detect",
      "Message": "Configuration overrides exist in 'E:\\opentabletdriver\\Configurations', overriding built-in configurations.",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806303+02:00",
      "Group": "Detect",
      "Message": "Searching for tablet 'Huion Kamvas Pro 16 (2.5K)'",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806346+02:00",
      "Group": "Device",
      "Message": "Initializing device 'Huion Tablet_GT1602' \\\\?\\hid#vid_256c&pid_006d&mi_00&col01#7&310b9f2e&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806387+02:00",
      "Group": "Device",
      "Message": "Using report parser type 'OpenTabletDriver.Configurations.Parsers.Huion.GianoReportParser'",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806429+02:00",
      "Group": "Detect",
      "Message": "Found tablet 'Huion Kamvas Pro 16 (2.5K)'",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806469+02:00",
      "Group": "IPC",
      "Message": "Connected to a client.",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.780651+02:00",
      "Group": "Display",
      "Message": "DPI Awareness enabled",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806551+02:00",
      "Group": "Huion Kamvas Pro 16 (2.5K)",
      "Message": "Output mode: Absolute Mode",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806593+02:00",
      "Group": "Huion Kamvas Pro 16 (2.5K)",
      "Message": "Display area: [5120x2880@<2560. 1440>:0°],",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806634+02:00",
      "Group": "Huion Kamvas Pro 16 (2.5K)",
      "Message": "Tablet area: [349,6x196,7@<174,8. 98,35>:0°],",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806675+02:00",
      "Group": "Huion Kamvas Pro 16 (2.5K)",
      "Message": "Clipping: Enabled",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806716+02:00",
      "Group": "Huion Kamvas Pro 16 (2.5K)",
      "Message": "Ignoring reports outside area: Disabled",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806757+02:00",
      "Group": "Huion Kamvas Pro 16 (2.5K)",
      "Message": "Tip Binding: [Mouse Button Binding: Left]@0%",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806798+02:00",
      "Group": "Settings",
      "Message": "Driver is enabled.",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.780684+02:00",
      "Group": "Display",
      "Message": "DPI Awareness enabled",
      "StackTrace": null,
      "Level": 0,
      "Notification": false
    },
    {
      "Time": "2022-08-21T02:21:41.7806881+02:00",
      "Group": "Settings",
      "Message": "Presets have been refreshed. Loaded 0 presets.",
      "StackTrace": null,
      "Level": 1,
      "Notification": false
    }
  ]
}

```

**Comment by jamesbt365** [comment]
Don't round the height and width, max sure the max x and y are correct in the `tablet > tablet debugger` check if the max pressure is correct and the tilt (if the tablet has any) works correctly. Also check if the tablet buttons and the pen button works then you can make a pull request and add this configuration if you would prefer to do that.

**Comment by DennisWG** [comment]
Yep, I've checked with the tablet debugger. Works fine. Also, I'm not rounding width and height. I've taken the values directly from Huion's product specification. Same with pressure for that matter.

**Comment by jamesbt365** [comment]
Spec sheets are often inaccurate, the correct width and height are 349.63x196.665

**Comment by jamesbt365** [comment]
fixed in #2410
